src/reanimated2/animations.js

```diff
@@ -33,7 +33,16 @@ export function cancelAnimation(value) {
 
 export function withTiming(toValue, userConfig, callback) {
   'worklet';
-  toValue = (typeof toValue === 'object' && toValue.value) ? toValue.value : toValue
+  // check toValue
+  const valueType = typeof toValue
+  if (valueType !== 'number') {
+    let error = `invalid type of toValue passed to withTiming, expected \`number\`, got \`${ valueType }\``;
+    if (valueType === 'object') {
+      error += ', maybe you forgot to add `.value`?'
+    }
+    throw new Error(error);
+  }
```

Comment:
Maybe it's worth extracting into some function?

Reply:
point well taken, fixed: 58f0e92